mypy/fscache.py

```diff
@@ -28,49 +28,45 @@
 advantage of the benefits.
 """
 
+import functools
 import os
 import stat
-from typing import Tuple, Dict, List, Optional
+from typing import Dict, List, Optional, Tuple, TypeVar
 from mypy.util import read_with_python_encoding
 
 
 class FileSystemMetaCache:
     def __init__(self) -> None:
-        self.flush()
+        self.stat = functools.lru_cache(maxsize=None)(self._stat)
```

Comment:
It's not clear to me why we are switching to `lru_cache` here -- I haven't followed all the previous discussion.

Reply:
I did it when I was trying to remove the memory leak. I can try switching back if you think that is better.

Reply:
I too was curious about traceback usage, so on that pull request [I tested the memory usage](https://github.com/python/mypy/pull/4809#issuecomment-376747428), and found no difference. I suppose the main argument for `lru_cache` is that it requires a bit less boilerplate, but I don't think that is super important. I will try replacing the `lru_cache`s with `dict`s.

Reply:
What's the current status of the memory leak investigation? Do you believe it may still be happening?